### ISSUE TITLE
[BUG] OpenTracing compatibility seems broken

### ISSUE THREAD

**Issue by OGKevin** [DataDog/dd-trace-php#1384]
### Bug description

When using Opentracing-php:v1.0.1, using ddtrace as tracer seems to not work. As the ddtrace implementation is not implementing the right interfaces.

With the following code:

```php
        // where tracer is \DDTrace\OpenTracer1\Tracer(); 
        $span = $this->tracer->startSpan("TEST_SPAN");
        $span->finish();
```

Triggers the following exception:

```
Argument 1 passed to DDTrace\OpenTracer1\Span::__construct() must implement interface DDTrace\Contracts\Span, instance of DDTrace\OpenTracer1\Span given, called in /opt/datadog-php/dd-trace-sources/src/DDTrace/OpenTracer1/Tracer.php on line 84
```

### PHP version

```
PHP 7.3.31-2+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Oct 22 2021 12:34:00) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.31, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.31-2+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v3.1.1, Copyright (c) 2002-2021, by Derick Rethans
    with ddtrace v0.67.0, Copyright Datadog, by Datadog
```
### Installed extensions

```
[PHP Modules]
bcmath
bz2
calendar
Core
ctype
curl
date
dba
ddtrace
dom
exif
fileinfo
filter
ftp
gd
gettext
hash
iconv
igbinary
imagick
intl
json
libxml
mbstring
msgpack
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
soap
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
wddx
xdebug
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Xdebug
Zend OPcache
ddtrace
```

### OS info

```
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
```

#### Output of phpinfo() (ddtrace >= 0.47.0)

Will provide this if its still needed.

**Comment by labbati** [comment]
This is unexpected (assuming that line 84 if the `startSpan` invocation, as it should be obvious). We have a test that runs that exact method - [link](https://github.com/DataDog/dd-trace-php/blob/master/tests/OpenTracer1Unit/TracerTest.php#L37).

Might I ask you to try our latest versions and, if it still fails, to provide a little bit more context about how you create the tracer and what's going on before that line?

**Comment by OGKevin** [comment]
> Might I ask you to try our latest versions
> with ddtrace v0.67.0, Copyright Datadog, by Datadog

Is this not the latest version based on https://github.com/DataDog/dd-trace-php/releases/tag/0.67.0? Or am i missing something?

My IDE says the same when I checkout the main branch of this project:
![Screenshot 2021-11-29 at 09 55 01](https://user-images.githubusercontent.com/17928966/143837053-41433d28-013d-4845-93c7-37727e863d5a.png)

Tracer is created the following way:
```php
    public function instantiate(DependencyInjector $di, string $requested_class): Tracer
    {
        if ($requested_class == MockTracer::class) {
            return new MockTracer();
        }
        if ($requested_class == Tracer::class) {
            return new \DDTrace\OpenTracer1\Tracer();
        }

        return new NoopTracer();
    }
```

Where `Tracer` is `OpenTracing\Tracer` using version `1.0.1`

**Comment by OGKevin** [comment]
Is there any other info that you need? This is a blocker.

**Comment by labbati** [comment]
@OGKevin I was finally able to look into this issue. This is totally a bug on our side and I was able to replicate your same error.

The issue is in this line of code https://github.com/DataDog/dd-trace-php/blob/aeb0f4fa6046041b8dce6d24aacdd48c527d75c7/src/DDTrace/OpenTracer1/Tracer.php#L32

The idea of the OT wrapper (`DDTrace\OpenTracer1\Tracer`) is to wrap our own DD tracer (`DDTrace\Tracer`, stored in `$this->tracer`), which is provided to the constructor. In your case you are not providing any constructor argument, and what goes into the `$this->tracer` property is the output of `self::make()`, which is a `DDTrace\OpenTracer1\Tracer` instead of the expected `DDTrace\Tracer`.

We will address this in the next release. In the meantime you can solve it by replacing
```
            return new \DDTrace\OpenTracer1\Tracer();
```
with
```
            return new \DDTrace\OpenTracer1\Tracer(\DDTrace\GlobalTracer::get());
```

Thanks for your patience.

**Comment by labbati** [comment]
The fix has been merged and will be released in 0.68.0
